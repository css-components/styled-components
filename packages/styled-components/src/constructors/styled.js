// @flow
import constructWithOptions from './constructWithOptions';
import StyledComponent from '../models/StyledComponent';
import domElements from '../utils/domElements';

import type { Target } from '../types';

const styled = (tag: Target) => constructWithOptions(StyledComponent, tag);

// Shorthands for all valid HTML Elements
for (let i = 0; i < domElements.length; i += 1) {
  styled[domElements[i]] = styled(domElements[i]);
}

export default styled;
