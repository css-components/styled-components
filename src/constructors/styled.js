// @flow
import css from './css'
import type { Interpolation, Target } from '../types'
import domElements from '../utils/domElements'
import enforceArray from '../utils/enforce-array-arguments'

export default (styledComponent: Function) => {
  const styled = (tag: Target) =>
    (strings: Array<string>, ...interpolations: Array<Interpolation>) => {
      const tagName = typeof tag === 'string' ? `.${tag}` : '(Component)'
      enforceArray(`styled${tagName}`)(strings)
      return styledComponent(tag, css(strings, ...interpolations))
    }

  // Shorthands for all valid HTML Elements
  domElements.forEach(domElement => {
    styled[domElement] = styled(domElement)
  })

  return styled
}
