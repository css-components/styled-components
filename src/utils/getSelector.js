// @flow
import isStyledComponent from './isStyledComponent';

function getSelector(chunk: any) {
  return isStyledComponent(chunk) ? `.${chunk.styledComponentId}` : chunk.toString();
}

export default getSelector;
