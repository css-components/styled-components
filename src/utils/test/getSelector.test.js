// @flow
import getSelector from '../getSelector';
import styled from '../../constructors/styled';

describe('getSelector', () => {
  it('returns a strinigified id for styled component', () => {
    const StyledComponent = styled.div`
      background-color: red;
    `;
    expect(getSelector(StyledComponent)).toMatchSnapshot();
  })

  it('returns a strinigified id for a nested styled component', () => {
    const StyledComponent = styled.div`
      background-color: purple;
    `;
    const NestedComponent = styled(StyledComponent)`
      color: blue;
    `;
    expect(getSelector(NestedComponent)).toMatchSnapshot();
  })

  it('returns name if not a styled component', () => {
    const StyledComponent = 'div'
  expect(getSelector(StyledComponent)).toBe('div')
  })
})
