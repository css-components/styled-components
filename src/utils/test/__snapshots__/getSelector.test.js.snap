// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSelector returns a strinigified id for a nested styled component 1`] = `".sc-htpNat"`;

exports[`getSelector returns a strinigified id for styled component 1`] = `".sc-bdVaJa"`;
